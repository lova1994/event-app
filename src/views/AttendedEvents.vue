<template>
  <div class="home">
attended events
<!-- <button @click="add">
Add
</button> -->

    <div class="container">
      <Events
        v-for="event in attendedEvents"
        :key="event.id"
        :event="event"
        
      />
      <!-- <h2>Attended Events</h2> -->
      <!-- <ul>
        <li v-for="event in attendedEvents" :key="event.id" :event="event">
          {{ event.title }}
        </li>
      </ul> -->
    </div>
  </div>
</template>

<script>
// @ is an alias to /src
import Events from "@/components/Events.vue";
import {  getAttendedEvents } from "@/data/events.js";

export default {
  name: "home",
  components: {
    Events,
  },
  data: () => ({
    attendedEvents: [],
  }),
  created() {
    this.attendedEvents = getAttendedEvents();
  },
  methods: {
  
  
 
  },
};
</script>




<style lang="scss">
@import "../assets/scss/_typography.scss";

#app {
  font-family: Mulish, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
   color: #000;
}

a {
  color: hotpink;
}

a:hover {
  font-weight: bold;
}
body {
  background-color: #fff;
  // display: flex;
  // // align-items: center;
  // justify-content: center;
  height: 100vh;
  margin: 0;
  font-family: muli;
  padding: 1em;
}

.container {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
  width: 940px;
  // margin:auto;
}

@media (max-width: 940px) {
  .container {
    grid-template-columns: repeat(2, 1fr);
    width: 200px;
  }
}
</style>


