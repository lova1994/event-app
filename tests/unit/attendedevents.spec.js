import { shallowMount, createLocalVue } from '@vue/test-utils'
import AttendedEvents from '@/views/AttendedEvents.vue'
import VueRouter from 'vue-router'


//testar
describe('Home.vue', () => {
    it('renders events', () => {
      const wrapper = shallowMount(AttendedEvents)
      const div = wrapper.find('div')
      expect(div.exists()).toBe(true)
    })
  })

  // snapshot test
test('mount a component', () => {
    const wrapper = shallowMount (AttendedEvents, {
    })
    expect(wrapper.html()).toMatchSnapshot()
    console.log(wrapper.html())
  })

  const localVue = createLocalVue()
localVue.use(VueRouter)

const router = new VueRouter()

shallowMount(AttendedEvents, {
  localVue,
  router
})

// check that router path works
test("test router", () => {
  const $route = {
    path: "/",
  };

  const wrapper = shallowMount(AttendedEvents, {
    mocks: {
      $route,
    },
    propsData: {
      single: "",
    },
  });
  expect(wrapper.vm.$route.path).toBe("/");
});


