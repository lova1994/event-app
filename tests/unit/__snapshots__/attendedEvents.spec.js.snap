// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount a component 1`] = `
<div class="home">
  attended events
  <div class="container">
    <events-stub event="[object Object]"></events-stub>
    <events-stub event="[object Object]"></events-stub>
  </div>
</div>
`;
